test/timezones/rounding.jl

```diff
@@ -0,0 +1,234 @@
+    @test round(zdt, Dates.Day) == ZonedDateTime(2015, 11, 1, tz)
+    @test floor(zdt, Dates.Hour) == prev_hour
+    @test ceil(zdt, Dates.Hour) == next_hour
+    @test round(zdt, Dates.Hour) == prev_hour
+    @test floor(zdt, Dates.Minute(30)) == prev_hour
+    @test ceil(zdt, Dates.Minute(30)) == between_hours
+    @test round(zdt, Dates.Minute(30)) == between_hours
+end
+
+###########################
+# ASIA/COLOMBO TRANSITION #
+###########################
+
+# Test rounding to ambiguous midnight, which (unfortunately) isn't handled well when
+# rounding to a DatePeriod resolution.
+
+zdt = ZonedDateTime(1996, 10, 25, 23, 55, colombo)  # 5 minutes before ambiguous half-hour
+@test floor(zdt, Dates.Day) == ZonedDateTime(1996, 10, 25, colombo)
+@test_throws AmbiguousTimeError ceil(zdt, Dates.Day)
+@test_throws AmbiguousTimeError round(zdt, Dates.Day)
+
+zdt = ZonedDateTime(1996, 10, 26, 0, 35, colombo)   # 5 minutes after ambiguous half-hour
+@test_throws AmbiguousTimeError floor(zdt, Dates.Day)
+@test_throws AmbiguousTimeError ceil(zdt, Dates.Day)    # floor is called during ceil
+@test_throws AmbiguousTimeError round(zdt, Dates.Day)
+
+# Rounding to the ambiguous midnight works fine using a TimePeriod resolution, however.
+
+zdt = ZonedDateTime(1996, 10, 25, 23, 55, colombo)  # 5 minutes before ambiguous half-hour
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 1)
+@test round(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 1)
+
+zdt = ZonedDateTime(1996, 10, 26, 0, 35, colombo)   # 5 minutes after ambiguous half-hour
+@test floor(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 2)
+
+# Rounding during the first half-hour between 00:00 and 00:30.
+
+zdt = ZonedDateTime(DateTime(1996, 10, 26, 0, 15), colombo, 1)
+@test floor(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 1)
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26, 0, 30), colombo)
+@test round(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 1)
+@test floor(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 1)
+@test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26, 0), colombo, 2)
+@test round(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26, 0), colombo, 2)
+
+# Rounding during the second half-hour between 00:00 and 00:30.
+
+zdt = ZonedDateTime(DateTime(1996, 10, 26, 0, 15), colombo, 2)
+@test floor(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 2)
+@test ceil(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26, 1), colombo)
+@test round(zdt, Dates.Hour) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 2)
+@test floor(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26), colombo, 2)
+@test ceil(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26, 0, 30), colombo)
+@test round(zdt, Dates.Minute(30)) == ZonedDateTime(DateTime(1996, 10, 26, 0, 30), colombo)
```

Comment:
You could specify all of these without making a `DateTime` first: `ZonedDateTime(1996, 10, 26, colombo, 2)`